sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+const { TransactionBlock } = require('@mysten/sui.js/transactions');
+const { saveConfig } = require('../evm/utils');
+const { Command } = require('commander');
+const { addBaseOptions } = require('./cli-utils');
+const { getWallet } = require('./sign-utils');
+const { printInfo } = require('../evm/utils');
+const chalk = require('chalk');
+const { loadSuiConfig } = require('./utils');
+
+const SUI_COIN_ID = '0x2::sui::SUI';
+
+class CoinManager {
+
+    static async getAllCoins(client, account) {
+        let cursor;
+        const coinTypeToCoins = {};
+
+        do {
```

Comment:
Create a generic util for paginated queries? It can take a lambda that accepts a cursor and makes the query (`async (cursor) => { return ...query... }`), and the util can iteratively call it and returns all items.

This is assuming that sui SDK uses `hasNextPage`, `nextCursor` and `cursor` fields for other paginated queries as well.